src/commands/ConvertDataSourceCommand.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as constants from '../constants';
+import * as extension from '../extension';
+import * as utils from '../utils';
+import * as vscode from 'vscode';
+import { DataVirtConfig, DataSourceConfig, Property, ConfigMapConfig, SecretConfig, MetaData, ConfigMapRef, KeyRef, SecretRef } from '../model/DataVirtModel';
+import { DataSourceTreeNode } from '../model/tree/DataSourceTreeNode';
+
+export async function convertDataSourceToSecret(dsNode: DataSourceTreeNode) {
+	await convertDataSource(dsNode, constants.REFERENCE_TYPE_SECRET);
+}
+
+export async function convertDataSourceToConfigMap(dsNode: DataSourceTreeNode) {
+	await convertDataSource(dsNode, constants.REFERENCE_TYPE_CONFIGMAP);
+}
+
+async function convertDataSource(dsNode: DataSourceTreeNode, refType: string) {
+
+	if(!dsNode.isEmpty() && !dsNode.isValueType()) {
+		vscode.window.showWarningMessage(`Datasource ${dsNode.dataSourceConfig.name} cannot be migrated to ${refType} because it is already a reference...`);
+		return;
+	}
+
+	const refName: string = await vscode.window.showInputBox( { validateInput: utils.validateFileNotExisting, placeHolder: 'Enter the name of the new reference.'});
+	if (refName === undefined) {
+		return;
+	}
+
+	const dvConfig: DataVirtConfig = dsNode.getProject().dvConfig;
+	if (dvConfig) {
+		const success: boolean = await createDataSource(dsNode.dataSourceConfig, refName, refType, dvConfig, dsNode.getProject().getFile());
```

Comment:
naming of this seems strange - are we just wrapping some of the convert code into a create?

Reply:
good catch. forgot to rename.